CorsixTH/Lua/world.lua

```diff
@@ -2264,13 +2263,24 @@ function World:removeLitter(obj, x, y)
 end
 
 --! Get the room at a given tile location.
---!param x X position of the queried tile.
---!param y Y position of the queried tile.
+--!param x (int) X position of the queried tile.
+--!param y (int) Y position of the queried tile.
 --!return (Room) Room of the tile, or 'nil'.
 function World:getRoom(x, y)
   return self.rooms[self.map:getRoomId(x, y)]
 end
 
+--! Get the hospital at a given tile location.
+--!param x (int) X position of the queried tile.
+--!param y (int) Y position of the queried tile.
+--!return (Hospital) Hospital at the given location or 'nil'.
+function World:getHospital(x, y)
+  for _, hosp in ipairs(self.hospitals) do
```

Comment:
This is a little roundabout, hosp:isInHospital calls a C++ function that tells it what hospital is at the current hospital and returns true if that hospital matches the hospital number. Because there are very few hospitals it doesn't matter, but this could be O(1) instead of O(N)